### PR TITLE
Fix parsing of '-' as positional argument

### LINKED ISSUE #120: A lone `-` should be treated as a regular positional argument
It is a common idiom to accept a `-` parameter meaning *stdin*. E.g., `cat -` reads its whole *stdin* and sends it to *stdout*. However, *argparse* currently treats it as a named parameter, even though `NamedArgument("")` is not allowed—so it cannot be accessed in any way.

```d
import argparse;

struct T {
    @NamedArgument bool c;
    @PositionalArgument(0) string fileName;
}

void main() {
    assert(CLI!T.parseArgs!((T t) { assert(t == T(true, "-")); })(["-", "-c"]) == 0);
}
```

### PR DESCRIPTION
**Pull request by andrey-zherikov**
Closes #120

### CODE REVIEW

File: source/argparse/internal/parser.d [hunk 1]
```diff
@@ -293,7 +293,7 @@ private Entry getNextEntry(bool bundling)(Config config, ref string[] args,
             }
         }
     }
-    else
+
```
Comment (SirNickolas) [inline source/argparse/internal/parser.d:296]: That would be amazing, but unfortunately, it’s not that simple.

```d
assert(!callParser!(enableStyling(Config.init, false), false)(t, ["-f"], unrecognizedArgs));
```

`-f` should not be bound to a positional argument.

File: source/argparse/internal/parser.d [hunk 2]
```diff
@@ -628,6 +620,38 @@ if(config.stylingMode == Config.StylingMode.autodetect)
         return callParser!(enableStyling(config, false), completionMode, COMMAND)(receiver, args, unrecognizedArgs);
 }
 
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+unittest
+{
+    import argparse.api.argument: PositionalArgument, NamedArgument;
+
+    struct T {
```
Comment (SirNickolas) [inline source/argparse/internal/parser.d:628]: I guess the brace should go on its own line, now that this snippet has become part of the codebase?

File: source/argparse/internal/parser.d (hunk 1 above)
Comment (SirNickolas) [reply source/argparse/internal/parser.d:296]: My guess is that we should return `else` where it was and move `arg0.length > 1` from line 147 to line 141, but I haven’t examined all branches carefully. It also has an implication that `---` (triple dash) will be rejected (GNU tools reject it as well so I personally don’t mind), which will affect the completer.

File: source/argparse/internal/parser.d (hunk 2 above)
Comment (andrey-zherikov) [reply source/argparse/internal/parser.d]: fixed

File: source/argparse/internal/parser.d (hunk 1 above)
Comment (andrey-zherikov) [reply source/argparse/internal/parser.d]: I agree with you - added this unit test and fixed

**Review by SirNickolas — APPROVED** [review]
`arg0.length > 1` in line 147 became redundant and can be removed now. Everything else looks good.

**Comment by andrey-zherikov** [comment]
> `arg0.length > 1` in line 147 became redundant and can be removed now. Everything else looks good.

Removed

**Pull request merged**